sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1474,6 +1474,34 @@ case class Pivot(
   override protected def withNewChildInternal(newChild: LogicalPlan): Pivot = copy(child = newChild)
 }
 
+/**
+ * A constructor for creating a transpose, which will later be converted
+ * to a [[LocalRelation]] at ReplaceTranspose during the query optimization.
+ *
+ * The result of the transpose operation is held in the `data` field, and the corresponding
+ * schema is stored in the `output` field. The `Transpose` node does not depend on any child
+ * logical plans after the data has been collected and transposed.
+ *
+ * @param output A sequence of output attributes representing the schema of the transposed data.
+ * @param data A sequence of [[InternalRow]] containing the transposed data.
+ * @param hasNonIndexColumns A flag indicating whether there are non-index columns in the
+ *                           original DataFrame. This is used to determine if the transposed data
+ *                           should be populated or if it should remain empty.
```

Comment:
How does this flag affect the final result?

Reply:
The flag is necessary to decide whether a Transpose node is resolved or not as shown below:

```scala
case class Transpose(...
  ...
  override lazy val resolved: Boolean = {
    (output.isEmpty && !hasNonIndexColumns) ||
      (output.nonEmpty && (!hasNonIndexColumns || data.nonEmpty))
  }
```

Reply:
Sorry I'm a bit confused. Shouldn't `Transpose` always be resolved? It's resolved from `UnresolvedTranspose`

Reply:
You are right! Removed `hasNonIndexColumns` and inherited "resolved`.